### PR TITLE
Suppress compiler warnings in release build.

### PR DESCRIPTION
**Pull request by reshke**
fix #ISSUE_Number

---

### Change logs

_Describe your change clearly, including what problem is being solved or what feature is being added._

_If it has some breaking backward or forward compatibility, please clary._

### Why are the changes needed?

_Describe why the changes are necessary._

### Does this PR introduce any user-facing change?

_If yes, please clarify the previous behavior and the change this PR proposes._

### How was this patch tested?

_Please detail how the changes were tested, including manual tests and any relevant unit or integration tests._

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to request `cloudberrydb/dev` team for review and approval when your PR is ready🥳

### CODE REVIEW

**Comment by my-ship-it** [comment]
Hey @reshke , the commit https://github.com/cloudberrydb/cloudberrydb/pull/673 already fix the compile issue.

File: src/backend/parser/parse_partition_gp.c [hunk 1]
```diff
@@ -1469,7 +1469,7 @@ generatePartitions(Oid parentrelid, GpPartitionDefinition *gpPartSpec,
 	 */
 	foreach(lc, gpPartSpec->partDefElems)
 	{
-		Node	   *n = lfirst(lc);
```
Comment (lss602726449) [inline src/backend/parser/parse_partition_gp.c:1472]: It seems that variable n is used in the next IsA(n, ColumnReferenceStorageDirective). Do we have need to add this PG_USED_FOR_ASSERTS_ONLY?

Comment (reshke) [reply src/backend/parser/parse_partition_gp.c:1472]: Looks like parts of this commit https://github.com/greenplum-db/gpdb-archive/commit/80576302ae02039c14c61d798b0631cb276a5790

oks, removed it

**Review by my-ship-it — APPROVED** [review]

**Comment by my-ship-it** [comment]
Please fix CI failures.

**Pull request merged**